### DISCUSSION TITLE
List the system requirements for self-hosting.

### DISCUSSION THREAD

**Discussion by johnnymatthews** [plausible/analytics#737] [Self-Hosted Support] [answered]
The [Getting Started guide](https://plausible.io/docs/self-hosting) doesn't say what the recommended system-specs are for a self-hosted version of Plausible.

I'm happy to create a draft PR, but I need to know what the requirements are first :smile:

**Comment by securitypedant** [comment]
I was just reviewing what it takes to self host a server and the lack of recommendations is painful. Would be great if someone can detail out min CPU, RAM.

**Comment by ukutaht** [comment, marked as answer]
The official answer is I don't know. We run on pretty large machines with 4-figure monthly spend. You probably don't want to do that.

I have a testing server that works fine with 1vCPU and 2GB ram ($10/mo from Digital Ocean).

How about everyone posts their CPU and RAM specs here so we can crowdsource the information?

**Reply by johnnymatthews** [reply]
Oh, nice! How well does that DO droplet run? Also, any idea of how much HDD space your installation takes up? Being able to spin up a $10 is handy, but not if we have to connect a $10 extra storage too.

**Reply by ukutaht** [reply]
Actually I just tested it today and it works fine for small time ranges but when I try to see the dashboard for a full year, it crashes the machine. My testing instance is running on the same data as our live demo: https://plausible.io/plausible.io?period=12mo

I'll try bumping it up to 4GB or something like that soon to see if it can handle the live demo. Realistically for most self-hosters the bottleneck will be RAM.

As for storage, here's a data point: https://github.com/plausible/analytics/discussions/1328#discussioncomment-1330111

**Reply by johnnymatthews** [reply]
> it crashes the machine

Ah, that's not ideal. Looks like HDD/SSD storage isn't a massive issue then. I'll also spin up a Digital Ocean node to see what the minimum RAM allowance is.

**Reply by ukutaht** [reply]
Haha it's not. Clickhouse is designed for machines with minimum 64GB ram but they've done some work recently to make it work on smaller machines. 2GB might still not be enough but I think 4 should be fine (I'll test it at some point).

**Reply by Vigasaurus** [reply]
Bonus data point here - for the same instance linked above re: storage space, it's on a DO 8GB, 4cpu droplet, but running alongside a relatively heavy node process that uses a consistent 3-4GB. So I'd assume 4GB is fine. And I can indeed on that day set do operations on the entire 14M events without much issue.

**Comment by MaluNoPeleke** [comment]
My hoster doesn't allow/offer Docker.
Is there an alternative to host it to Docker?
What about a simple zip package that can be installed similarly to Matomo?
